PEP621 and migrate latest python

antirais opened the pull request:
This PR will:
* add pyproject.toml (PEP621) support
* will support python3.13
* upgrade deprecated libraries (resolves #196, #197)

Tested this build on Kali 2025.2

ZanyMonk commented:
Hey, thanks for the PR. At first glance everything looks good.

Have you tried running the automated tests ?

I see a lot of cosmetic changes, did you change anything important while moving `weevely.py` or something ?

Would you mind reverting [this commit](https://github.com/epinna/weevely3/pull/200/commits/4e66cee6727591597cc75528369be2acb2e21725) please ? I think the urls should keep pointing at here as long as it is the "official" repo.

I won't be around for a couple of weeks but I'll make sure to give this a lil more time to review and merge.

antirais commented:
Hey!

> Have you tried running the automated tests ?

I did, but they are failing, and I didn't bother with it at the time.
For latest weevely3 4.0.2 release, I saw:
```sh
Ran 85 tests in 24.139s

FAILED (errors=78, skipped=2)
```

I did patch them a bit in my branch and committed the changes so that I could at least compare.
This is the latest result. I only patched imports and set up a bit to make it run, needs a clean-up.

```sh
----------------------------------------------------------------------
Ran 85 tests in 73.066s

FAILED (failures=2, errors=1, skipped=2)
```

I tested my fork manually with some obfuscator generators, and I was able to use it as expected.

> I see a lot of cosmetic changes, did you change anything important while moving weevely.py or something ?

I tried to keep the changes to the functionality to the minimum. Updated some dependencies and fixed some issues with latest python version. The main change is that weevely is installed as a script and uses `src/` layout. As a user, I can now run `uv tool install https://github.com/antirais/weevely.git; weevely -h`.

> Would you mind reverting [this commit](https://github.com/epinna/weevely3/pull/200/commits/4e66cee6727591597cc75528369be2acb2e21725) please ? I think the urls should keep pointing at here as long as it is the "official" repo.

Sure, no problem. 

> I won't be around for a couple of weeks but I'll make sure to give this a lil more time to review and merge when I'm back.

Sounds good!

ZanyMonk commented:
Hey,
better late than never as they say ! Let's be honest, I totally forgot about your PR, apologies.

Apart from a few very minor changes (urls to your repo in pyproject.toml and duplicated lines in check.py) everything looks good to me.

All tests pass except two of them (`test_net_proxy` and `test_shell_ssh`).

Could you allow me to edit this PR please ? So I can push a fix for these issues myself. There should be a checkbox labelled `Maintainers are allowed to edit this pull request` on this page.

Thank you again for this quality PR ! Sorry it took me so long to review it.

ZanyMonk commented:
Aaaaaand I f*cked up my push ...

@antirais you should be able to reset your repo to this commit : ba0ca50a3746397ad25bb6a26a1229cb0512fabf
I cannot do it myself, Github allowed me to mess up with your master branch but doesn't allow me to fix my mistake.

I created [another PR](https://github.com/epinna/weevely3/pull/202) with your changes, just in case.
